src/main/java/sg/dex/starfish/impl/remote/RemoteJob.java

```diff
@@ -37,7 +37,11 @@ public boolean isDone() {
     @Override
 	@SuppressWarnings("unchecked")
 	public synchronized Map<String,Object> pollResult() {
-        if (response != null) return response;
+        if (response != null) {
+            this.status= Status.succeeded;
```

Comment:
Is this correct? What if the response indicates a remote failure? Needs good tests.

Also I think "pollResult()" needs to do a remote HTTP call to get the latest status?